src/solver/hydro/management/daily.cpp

```diff
@@ -112,104 +117,89 @@ struct DebugData
-                                 << '\t' << '\t' << '\t'
-                                 << (data.MOG[realmonth] / reservoirCapacity
-                                     + previousMonthWaste[realmonth])
-                                      * 100;
-                        }
-                        file << '\n';
-
-                        dayMonth++;
-                    }
+            buffer << "\n";
+            buffer << "-------------\n";
+            buffer << monthName.c_str() << "\n";
+            buffer << "-------------\n";
+            buffer << "\t\t\tNiveauMin\tApports\t\tTurbMax\t\tTurbCible\tTurbCible "
+                      "MAJ\tNiveaux D\tNiveaux F\tTurbines\t";
+            buffer << "Overflows\tDeviations\tViolations\tDeviation Max\tViolation "
+                      "Max\tWaste\t\tCout total\tTurb mois no previous W\t\tTurb mois + "
+                      "previous W \n";
+
+            uint dayMonth = 1;
+            for (uint day = firstDay; day != endDay; ++day)
+            {
+                double turbines = valgen.HydrauliqueModulableQuotidien[day] / reservoirCapacity;
+                double niveauDeb = valgen.NiveauxReservoirsDebutJours[day];
+                double niveauFin = valgen.NiveauxReservoirsFinJours[day];
+                double apports = srcinflows[day] / reservoirCapacity;
+                double turbMax = maxP[day] * maxE[day] / reservoirCapacity;
+                double turbCible = dtg[day] / reservoirCapacity;
+                double turbCibleUpdated
+                  = dtg[day] / reservoirCapacity + previousMonthWaste[realmonth] / daysPerMonth;
+                buffer << day << '\t' << '\t' << dayMonth << '\t' << lowLevel[day] * 100 << '\t'
+                       << apports * 100 << '\t' << turbMax * 100 << '\t' << turbCible * 100 << '\t'
+                       << turbCibleUpdated * 100 << '\t' << '\t' << niveauDeb * 100 << '\t'
+                       << niveauFin * 100 << '\t' << turbines * 100 << '\t' << OVF[day] * 100
+                       << '\t' << DEV[day] * 100 << '\t' << VIO[day] * 100;
+                if (dayMonth == 1)
+                {
+                    buffer << '\t' << deviationMax[realmonth] * 100 << '\t' << '\t'
+                           << violationMax[realmonth] * 100 << '\t' << '\t'
+                           << WASTE[realmonth] * 100 << '\t' << CoutTotal[realmonth] << '\t'
+                           << (data.MOG[realmonth] / reservoirCapacity) * 100 << '\t' << '\t'
+                           << '\t' << '\t' << '\t'
+                           << (data.MOG[realmonth] / reservoirCapacity
+                               + previousMonthWaste[realmonth])
+                                * 100;
                 }
+                buffer << '\n';
+
+                dayMonth++;
             }
-            file.close();
         }
+        auto buffer_str = buffer.str();
+        pWriter->addJob(path.str(), buffer_str);
```

Comment:
why not directly : 
```C++
        pWriter->addJob(path.str(), buffer.str());
```
instead of : 
```C++
        auto buffer_str = buffer.str();
        pWriter->addJob(path.str(), buffer_str);
```
?

Reply:
Then why not have a **const** qualifier for 2nd parameter of **addJob**, like it's done for the 1st ? 
I guess the **content** is not modified by **addJob** anyway